src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -107,6 +107,14 @@ trait SubmissionComponent {
       }))
     }
 
+    def countByStatus(workspaceContext: SlickWorkspaceContext): ReadAction[Map[String, Int]] = {
+      filter(_.workspaceId === workspaceContext.workspaceId).groupBy(s => s.status).map { case (status, submissions) =>
```

Comment:
yeah, it's a lot of Slick magic syntax. Because countByStatus is part of the submissionQuery object, and submissionQuery defines itself on the SUBMISSION db table, Slick knows this is filtering on that table. "_.workspaceId" equates to the SUBMISSION.WORKSPACE_ID column.